src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -833,6 +839,39 @@ private static ulong CurlSendCallback(IntPtr buffer, ulong size, ulong nitems, I
                 return Interop.Http.CURL_READFUNC_ABORT;
             }
 
+            private static void SetLowSpeedLimit(EasyRequest easy, TimeSpan timeout)
+            {
+                // If there's no send/receive timeout on the handler, the defaults will
+                // remain disabled, and we don't need to worry about setting or resetting.
+                if (!easy._handler._hasSendReceiveTimeout)
+                {
+                    return;
+                }
+
+                // We need to ensure the timeout is set appropriately, which may include
+                // disabling a timeout currently set if one was.
+
+                long limit = 0, time = 0; // 0 == disable
+
+                if (timeout != Timeout.InfiniteTimeSpan)
+                {
+                    double secs = timeout.TotalSeconds;
+                    if (secs >= 1)
+                    {
+                        limit = 1;
```

Comment:
> Send the GET and request headers. Wait unbounded.

I probably need to set the timeout after all request content data has been sent, or if there is none, set it immediately when initiating the request.

> If the data rate stays below 1 byte per second for 30 seconds, abort, otherwise read forever.

Yes.  Isn't that basically the same as saying "if we don't receive a byte for 30 seconds, abort"?  That's what the timeout is meant to be: if we go N seconds without getting the data requested, abort.

> Except we already read, since curl already told us how big stuff is. So maybe it's the timeout for the next header?

It's meant to be the timeout for each underlying send/receive.  Do you see a better way to achieve it?

> Repeat for every header.

I'm only calling the set method on each header because it's the easiest place to hook it.  If the speed limit information hasn't changed, it's effectively a nop.